src/core/modbus-client-core.js

```diff
@@ -236,8 +242,16 @@ de.biancoroyal.modbus.core.client.writeModbusByFunctionCodeFive = function (node
   node.client.writeCoil(parseInt(msg.payload.address), msg.payload.value).then(function (resp) {
     coreClient.activateSendingOnSuccess(node, cb, cberr, resp, msg)
   }).catch(function (err) {
-    coreClient.activateSendingOnFailure(node, cberr, err, msg)
-    node.modbusErrorHandling(err)
+    if (node.client.getID() === 0) {
+      const resp = {
+        address: parseInt(msg.payload.address),
+        value: parseInt(msg.payload.value)
+      }
+      coreClient.activateSendingOnSuccess(node, cb, cberr, resp, msg)
+    } else {
+      coreClient.activateSendingOnFailure(node, cberr, err, msg)
+      node.modbusErrorHandling(err)
+    }
```

Comment:
Some clarification here: modbus-serial does return an error when a response is not received, even if unitid=0. See [this issue](https://github.com/yaacov/node-modbus-serial/issues/97) Therefore I added the check in the error handler..